### PR TITLE
internal/charmstore: get upload info parts end point.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2087/

**Review by jrwren — APPROVED** [review]

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2088/

File: internal/v5/upload.go [hunk 1]
```diff
@@ -132,6 +132,31 @@ func (h *ReqHandler) serveUploadPart(w http.ResponseWriter, req *http.Request) e
 		default:
 			return errgo.WithCausef(nil, params.ErrNotFound, "")
 		}
+	case "GET":
+		elems := strings.Split(strings.TrimPrefix(req.URL.Path, "/"), "/")
+		if len(elems) != 1 {
+			return badRequestf(nil, "unknown GET method called")
```
Comment (rogpeppe) [inline internal/v5/upload.go]: I'm thinking this should probably be a 404.

File: internal/v5/upload.go [hunk 2]
```diff
@@ -132,6 +132,31 @@ func (h *ReqHandler) serveUploadPart(w http.ResponseWriter, req *http.Request) e
...
+		}
+		uploadId := elems[0]
+		uploadInfo, err := h.Store.BlobStore.UploadInfo(uploadId)
+		if err != nil {
+			return errgo.Mask(err, errgo.Is(params.ErrNotFound))
```
Comment (rogpeppe) [inline internal/v5/upload.go]: Let's have a test for this case, please.

**Review by rogpeppe — APPROVED** [review]
LGTM modulo some comments and suggestions. Also, it would be great to have this endpoint added to the API doc.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2089/

**Comment by fabricematrat** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmstore-merge

**Pull request merged**